src/tool-classes.ts

```diff
@@ -0,0 +1,179 @@
+            await githubIssue.createIssue(this.internalRepository, this.name, versionClosestToEol.latest);
+            await githubIssue.sendIssueToSlack(this.name, versionClosestToEol.latest);
+            return;
+        }
+    
+        core.info(`The version of ${this.name} provided by the API (${versionClosestToEol.latest}) matches the one in the manifest (${earliestVersionInManifest}). Checking the EOL support date...\n`);
+    
+        if (isDateMoreThanSixMonthsAway(new Date(versionClosestToEol.eol))) {
+            core.info(`${this.name} version ${versionClosestToEol.latest} has more than 6 months left before EOL. It will reach its EOL date on ${versionClosestToEol.eol} \n`);
+            return;
+        }
+        
+        const issueContent = {
+            title: `[AUTOMATIC MESSAGE] ${this.name} version \`${versionClosestToEol.latest}\` is losing support on ${versionClosestToEol.eol}`,
+            body:  `Hello :wave: 
+                    The support for ${this.name} version \`${versionClosestToEol.latest}\` is ending on ${versionClosestToEol.eol}. Please consider upgrading to a newer version of ${this.name}.`,
+            labels: ['deprecation-notice'],
+        };
+
+        const githubIssue = new GitHubIssue(issueContent.title, issueContent.body, issueContent.labels);
+        await githubIssue.createIssue(this.internalRepository, this.name, versionClosestToEol.latest);
+        await githubIssue.sendIssueToSlack(this.name, versionClosestToEol.latest);
+        return;
+        
+    }
+}
+
+
+export class NodeTool extends Tool {
+    constructor(
+        name: string = 'Node', 
+        eolApiEndpoint: string = 'https://endoflife.date/api/node.json', 
+        manifestRepository: ManifestRepository = new ManifestRepository('actions', 'node-versions')
+        ) {
+        super(name, eolApiEndpoint, manifestRepository);
+    }
+}
+
+
+export class PythonTool extends Tool {
+    constructor(
+        name: string = 'Python',
+        eolApiEndpoint: string = 'https://endoflife.date/api/python.json',
+        manifestRepository: ManifestRepository = new ManifestRepository('actions', 'python-versions')
+        ) {
+        super(name, eolApiEndpoint, manifestRepository);
+    }
+}
+
+
+export class GoTool extends Tool {
+    constructor(
+        name: string = 'Go',
+        eolApiEndpoint: string = 'https://endoflife.date/api/go.json',
+        manifestRepository: ManifestRepository = new ManifestRepository('actions', 'go-versions')
+        ) {
+        super(name, eolApiEndpoint, manifestRepository);
+    }
+
+    async checkVersions() {
+        const goVersionsFromEolApi = await this.getVersionsFromApi(this.eolApiEndpoint);
+        const firstTwoVersionsFromEolApi =  goVersionsFromEolApi.slice(0, 2);
+        const reversedFirstTwoVersions = firstTwoVersionsFromEolApi.reverse();
+        const versionClosestToEol = reversedFirstTwoVersions[0];
```

Comment:
Why do we need this complex logic? Let's get the element we need right away and leave a comment why why this works this way
```suggestion
        // Each major Go release is supported until there are two newer major releases
        // The security policy can be found at https://go.dev/doc/devel/release#policy
        const versionClosestToEol =  goVersionsFromEolApi[1];
```

Reply:
Fixed to say this instead